cli/azd/internal/appdetect/maven.go

```diff
@@ -0,0 +1,206 @@
+package appdetect
+
+import (
+	"archive/zip"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+func getMvnCommand(pomPath string) (string, error) {
+	mvnwCommand, err := getMvnwCommandInProject(pomPath)
+	if err == nil {
+		return mvnwCommand, nil
+	}
+	if commandExistsInPath("mvn") {
+		return "mvn", nil
+	}
+	return getDownloadedMvnCommand()
+}
+
+func getMvnwCommandInProject(pomPath string) (string, error) {
+	mvnwCommand := "mvnw"
+	dir := filepath.Dir(pomPath)
+	for {
+		commandPath := filepath.Join(dir, mvnwCommand)
+		if fileExists(commandPath) {
+			return commandPath, nil
+		}
+		parentDir := filepath.Dir(dir)
+		if parentDir == dir {
+			break
+		}
+		dir = parentDir
+	}
+	return "", fmt.Errorf("failed to find mvnw command in project")
+}
+
+const mavenVersion = "3.9.9"
+const mavenURL = "https://repo.maven.apache.org/maven2/org/apache/maven/apache-maven/" +
+	mavenVersion + "/apache-maven-" + mavenVersion + "-bin.zip"
```

Comment:
Move this out of func to solve this problem:
https://github.com/azure-javaee/azure-dev/actions/runs/12353956134/job/34474195291
> ![image](https://github.com/user-attachments/assets/2a592ef2-9328-48c6-9d4d-95105a0994bd)